opencti-platform/opencti-graphql/src/migrations/1716815108224-update-data-sharing-ingestion-capabilities.js

```diff
@@ -0,0 +1,82 @@
+import { logApp } from '../config/conf';
+import { executionContext, SYSTEM_USER } from '../utils/access';
+import { elList, elLoadById, elReplace } from '../database/engine';
+import { roleCapabilities } from '../domain/user';
+import { addCapability } from '../domain/grant';
+import { createRelation } from '../database/middleware';
+import { ENTITY_TYPE_ROLE } from '../schema/internalObject';
+import { READ_INDEX_INTERNAL_OBJECTS } from '../database/utils';
+
+const message = '[MIGRATION] Split data sharing & ingestion into 2 capabilities';
+
+export const up = async (next) => {
+  logApp.info(`${message} > started`);
+  const context = executionContext('migration');
+
+  // ------ Create Access ingestion
+  const accessIngestionCapability = await addCapability(context, SYSTEM_USER, {
+    name: 'INGESTION',
+    attribute_order: 2600,
+    description: 'Access ingestion'
+  });
+  // ------ Create Manage ingestion
+  const manageIngestionCapability = await addCapability(context, SYSTEM_USER, {
+    name: 'INGESTION_SETINGESTIONS',
+    description: 'Manage ingestion',
+    attribute_order: 2610
+  });
+
+  // ------ Update roles
+  const callback = async (roles) => {
+    for (let i = 0; i < roles.length; i += 1) {
+      const roleId = roles[i].id;
+      const capabilities = await roleCapabilities(context, SYSTEM_USER, roleId);
+      // Select 'Access ingestion' if 'Access Data sharing & ingestion' is selected
+      const hasAccessDataSharingCapability = capabilities.some((capability) => capability.name === 'TAXIIAPI');
+      if (hasAccessDataSharingCapability) {
```

Comment:
so here for roles with TAXIIAPI capability, we add INGESTION capability, that's right ? 

Shouldn't we add it too to the default "connectors" role since it has TAXIIAPI capability ?

Reply:
Done